radicale/storage/multifilesystem/__init__.py

```diff
@@ -83,6 +84,15 @@ def _atomic_write(self, path, mode="w", newline=None, sync_directory=True,
                 raise RuntimeError("Fsync'ing file %r failed: %s" %
                                    (path, e)) from e
             tmp.close()
+            if os.path.exists(path):
+                os.chmod(tmp.name, os.stat(path).st_mode)
+            else:
+                tmp_mode = os.stat(tmp.name).st_mode
+                current_mask = os.umask(0)
+                os.umask(current_mask)
+                if not (current_mask & stat.S_IRGRP):
+                    tmp_mode |= stat.S_IRGRP
+                    os.chmod(tmp.name, tmp_mode)
```

Comment:
The new mode should always be set to ``0o666 & ~current_mask`` to match the behavior of creating a file with ``open(...)``.